src/common/collections.rs

```diff
@@ -659,6 +659,26 @@ pub async fn do_update_collection_cluster(
                 )
                 .await
         }
+        ClusterOperations::FinishResharding(_) => {
+            // TODO(resharding): Deduplicate resharding operations handling?
+
+            let Some(state) = collection.resharding_state().await else {
+                return Err(StorageError::bad_request(format!(
+                    "resharding is not in progress for collection {collection_name}"
+                )));
+            };
```

Comment:
It would be good to add another condition check here. Just checking for the committed write hash ring is probably good enough for now.

Reply:
It's done when applying consensus operation anyway.

Reply:
Good point. That's good enough.